tests/e2e/taqueria-registered-contracts.spec.ts

```diff
@@ -1,183 +1,159 @@
+		expect(stdout).toEqual(expect.arrayContaining(['List registered contracts']));
+
+		await cleanup();
 	});
 
-	test('Initial config has no registered contracts', async () => {
-		const config = await readFile(configFile, { encoding: 'utf-8' });
-		const json = JSON.parse(config);
+	test('Initial config will have no registered contracts', async () => {
+		const { execute, cleanup, readFile, exists } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
+
+		const json = JSON.parse(await readFile('./test-project/.taq/config.json'));
 		expect(json).toBeInstanceOf(Object);
 		expect(json).not.toHaveProperty('contracts');
-	});
 
-	test('Assure that a non-existing contract cannot be registered', async () => {
-		const nonExistingContract = join(taqueriaProjectPath, 'contracts', 'nonexisting');
-		const result = exec('taq add-contract nonexisting', { cwd: taqueriaProjectPath });
-		expect(result).rejects.toHaveProperty('stderr');
-		try {
-			await result;
-		} catch (err) {
-			const { stderr } = err as { stderr: string };
-			expect(stderr).toContain('Could not read');
-			expect(stderr).toContain(nonExistingContract);
-		}
-
-		const config = await readFile(configFile, { encoding: 'utf-8' });
-		const json = JSON.parse(config);
-		expect(json).toBeInstanceOf(Object);
-		return expect(json).not.toHaveProperty('contracts');
+		await cleanup();
 	});
 
-	test('Assure that an existing contract can be registered', async () => {
-		await copyFile(
-			'e2e/data/hello-tacos.mligo',
-			join(taqueriaProjectPath, 'contracts', 'hello-tacos.mligo'),
-		);
-		const result = exec('taq add-contract hello-tacos.mligo', { cwd: taqueriaProjectPath });
-		expect(result).resolves.toHaveProperty('stdout');
-		await result;
+	test('add-contract will not register a contract that does not exist', async () => {
+		const { execute, cleanup, readFile, exists } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
 
-		const bytes = await readFile(join(taqueriaProjectPath, 'contracts', 'hello-tacos.mligo'));
-		const digest = createHash('sha256');
-		digest.update(bytes);
-		const hash = digest.digest('hex');
+		const { stderr } = await execute('taq', 'add-contract no_such_contract', './test-project');
+		expect(stderr).toEqual(expect.arrayContaining(['Could not read {{base}}/test-project/contracts/no_such_contract']));
 
-		const config = await readFile(configFile, { encoding: 'utf-8' });
-		const json = JSON.parse(config);
+		const json = JSON.parse(await readFile('./test-project/.taq/config.json'));
 		expect(json).toBeInstanceOf(Object);
-		expect(json).toHaveProperty('contracts');
-		return expect(json.contracts).toEqual({
-			'hello-tacos.mligo': {
-				sourceFile: 'hello-tacos.mligo',
-				hash,
-			},
-		});
+		expect(json).not.toHaveProperty('contracts');
```

Comment:
Hmmm this is testing implementation

Reply:
Removed the unnecessary implementation stuff. There are a couple of cases where the config file has to be set up before the test; I had to leave those.